app/workers/scheduler/follow_recommendations_scheduler.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class Scheduler::FollowRecommendationsScheduler
+  include Sidekiq::Worker
+  include Redisable
+
+  sidekiq_options retry: 0
+
+  def perform
+    fallback_recommendations = FollowRecommendation.safe.filtered.limit(100)
+
+    I18n.available_locales.each do |locale|
+      recommendations = []
+
+      # We can skip the work if no accounts with that language exist
+      recommendations.concat(FollowRecommendation.safe.filtered.localized(locale).limit(100)) if AccountSummary.safe.filtered.localized(locale).exists?
+
+      # Use language-agnostic results if there are not enough language-specific ones
+      recommendations.concat(fallback_recommendations.take(100 - recommendations.size)) if recommendations.size < 100
+
+      redis.pipelined do
+        redis.del(key(locale))
+
+        recommendations.each do |recommendation|
+          redis.zadd(key(locale), recommendation.rank, recommendation.account_id)
+        end
+      end
```

Comment:
I was wondering why the ordering was mostly unchanged between `fr` and `en` recommendations on my instance. The reason is that the recommendation rank is mixed in with no regard with the matching language. Maybe this needs changing.

Furthermore, a recommendation set of 100 accounts seems like a lot, but idk.

Reply:
I see. It's unintentional but I can see it working both ways. You could argue that a globally higher rank account might be more interesting than a language-specific one, but you could also bump up language-specific ones by adding +1 to their score.

If I'm not mistaken you can request up to 80 accounts per page from the API. 20 more accounts leaves some padding for results getting filtered out by blocks/mutes/domain blocks.

Reply:
It seems you can request up to 80 accounts indeed, but there is no paging possible. Anyway, that still seems a lot but that's not really an issue.

Not completely sure about the exact bump to the rank that should happen. Also, I just noticed that there is no explicit duplicate check, which isn't an issue right now, but will be if we bump the score based on the language.

Reply:
No, but it's likely that the language-specific recommendations and the generic ones do overlap. And you don't exclude the language-specific recommendations from the generic set before taking accounts from the generic set., so you'll end up with duplicates, with the generic ones overwriting the language-specific ones in the resulting set.

Reply:
I don't think their scores would be different, the language would affect which accounts are returned but not how their scores are calculated

Reply:
Except if you bump the score for matching-language suggestions, as we were discussing.

Reply:
At this stage I don't know what delivers better results so I'll leave it this way and it can be changed later.